src/main/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiImpl.java

```diff
@@ -74,12 +83,46 @@ public ProfileEntitlementResponse getProfileEntitlements(String bearerToken, Int
     }
 
     @Override
-    public MasterIndexFieldByRecordClassSchemaResponse getMasterIndexFieldByRecordClassSchema(String bearerToken,
-                                                                                              Integer executionId,
-                                                                                              Integer rpoStageId, UserAccountEntity userAccount) {
-        throw new NotImplementedException("Method not implemented");
+    public List<MasterIndexFieldByRecordClassSchema> getMasterIndexFieldByRecordClassSchema(String bearerToken,
+                                                                                            Integer executionId,
+                                                                                            ArmRpoStateEntity rpoStageId,
```

Comment:
Should the method only ever be called with a "state" corresponding to `GET_MASTERINDEXFIELD_BY_RECORDCLASS_SCHEMA_PRIMARY` or `GET_MASTERINDEXFIELD_BY_RECORDCLASS_SCHEMA_SECONDARY`, or do we allow any state?

If we're limiting it, consider adding a validation check and throw an exception if any other state is passed.

Reply:
@hemantasharma1129 could you clarify the expectations on this one?

Reply:
@chrisbellingham-hmcts As of now, this method is only meant to accept those 2 states. The reason for introducing 2 states for the same endpoint is because the endpoint will be called twice in the entire End to End ARM Rpo flow. I agree, the parameter name should be changed to match the Entity. @karen-hedges FYI